api/endpoints/internal.py

```
@@ -5,41 +5,121 @@
+        # Step 1: Parse document text
+        text = None
+
+        import base64
+        import binascii
+        tmp_file = None
+        try:
+            if request.file_type == "pdf":
+                with tempfile.NamedTemporaryFile(delete=False, suffix=".pdf", mode="wb") as tmp:
+                    try:
+                        data = base64.b64decode(request.source_material)
+                    except (binascii.Error, ValueError) as decode_err:
```

Ruff (0.12.2)

Local variable `decode_err` is assigned to but never used

Remove assignment to unused variable `decode_err`

(F841)